src/core/Akka/Actor/ActorCell.FaultHandling.cs

```diff
@@ -348,7 +351,9 @@ private void FinishRecreate(Exception cause, ActorBase failedActor)
 
                 if (System.Settings.DebugLifecycle)
                     Publish(new Debug(_self.Path.ToString(), freshActor.GetType(), "Restarted (" + freshActor + ")"));
-
+                if(System.Settings.EmitActorTelemetry)
+                    System.EventStream.Publish(new ActorRestarted(Self, Props.Type, cause.ToReason()));
```

Comment:
Emit `ActorRestarted` events here automatically when an actor is being rebooted. Include details about the exception that caused it to restart (so that data can be aggregated downstream.)